_includes/files.html

```diff
@@ -32,32 +32,50 @@
 {% for file in files %}
 
   {% comment %}
-    Determine correct collection based on language param.
+    Determine which page includes this file. It will match the name of the
+    folder that contains the file, for example:
+
+      _files/sia-nextcloud/Dockerfile.sia
+             ^^^^^^^^^^^^^
+             Matching page slug
   {% endcomment %}
-  {% assign file_path = file.relative_path  | remove: '_files/'  | remove: '_ymls/' %}
-  {% if include.language == "yml" or include.language == "yaml" %}
-    {% assign file_path = file_path | append: "." | append: include.language %}
+  {% assign file_path_parts = file.relative_path | split: '/' %}
+  {% assign matching_page_slug = file_path_parts[1] %}
+
+  {% comment %}
+    If we're not in the page that includes this file, skip it.
+  {% endcomment %}
+  {% if matching_page_slug != page.slug %}
+    {% continue %}
   {% endif %}
-  {% assign link = site.baseurl | append: '/files/' | append: file_path %}
 
   {% comment %}
-    Determine file's subdirectory to compare to post subdirectory.
+    Determine the relative URL of the file as it will appear on the site.
   {% endcomment %}
-  {% assign subdir_array = file.path | split: '/' %}
-  {% assign subdir = subdir_array[1] %}
+  {% assign file_relative_url = file.relative_path | replace: '_files/', '/files/' | replace: '_ymls/', '/files/' %}
+  {% comment %}
+    YAML files are a special case because the files are not stored with their
+    extensions. We need to add the extensions back, so a URL like:
 
-  {% if subdir == page.slug %}
+      /files/sia-nextcloud/docker-compose
+
+    becomes:
+
+      /files/sia-nextcloud/docker-compose.yml
+  {% endcomment %}
+  {% if include.language == 'yml' or include.language == 'yaml' %}
+    {% assign file_relative_url = file_relative_url | append: '.' | append: include.language %}
```

Comment:
I can update this in the subsequent task for this code, but upon review with fresh eyes, this needs to change and I want to make sure I make a not of it.  I think the last append here should be hardcoded to "yml" because if the include.language is set to "yaml" then the embed of the file fails.

So this line should more simply become:
`{% assign file_relative_url = file_relative_url | append: '.yml' %}`

Reply:
Ah, good catch. Fixed in 93bf0d2801041c8706e865d176f9acca3a6d9512.